src/code/z_play.c

```diff
@@ -110,34 +110,254 @@
+}
+
+void Play_SetupRespawnPoint(GameState* gameState, s32 respawnMode, s32 playerParams) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (globalCtx->sceneNum != SCENE_KAKUSIANA) { // Grottos
+        Play_SetRespawnData(&globalCtx->state, respawnMode, (u16)((void)0, gSaveContext.entranceIndex),
+                            globalCtx->roomCtx.currRoom.num, playerParams, &player->actor.world.pos,
+                            player->actor.shape.rot.y);
+    }
+}
+
+// Override respawn data in Sakon's Hideout
+void func_80169ECC(GlobalContext* globalCtx) {
+    if (globalCtx->sceneNum == SCENE_SECOM) {
+        globalCtx->nextEntranceIndex = 0x2060;
+        gSaveContext.respawnFlag = -7;
+    }
+}
+
+// Gameplay_TriggerVoidOut ?
+// Used by Player, Ikana_Rotaryroom, Bji01, Kakasi, LiftNuts, Test4, Warptag, WarpUzu, Roomtimer
+void func_80169EFC(GameState* gameState) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+
+    gSaveContext.respawn[0].tempSwchFlags = globalCtx->actorCtx.flags.switches[2];
+    gSaveContext.respawn[0].unk_18 = globalCtx->actorCtx.flags.collectible[1];
+    gSaveContext.respawn[0].tempCollectFlags = globalCtx->actorCtx.flags.collectible[2];
+    globalCtx->nextEntranceIndex = gSaveContext.respawn[0].entranceIndex;
+    gSaveContext.respawnFlag = 1;
+    func_80169ECC(globalCtx);
+    globalCtx->sceneLoadFlag = 0x14;
+    globalCtx->unk_1887F = 2;
+}
+
+// Gameplay_LoadToLastEntrance ?
+// Used by game_over and Test7
+void func_80169F78(GameState* gameState) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+
+    globalCtx->nextEntranceIndex = gSaveContext.respawn[2].entranceIndex;
+    gSaveContext.respawnFlag = -1;
+    func_80169ECC(globalCtx);
+    globalCtx->sceneLoadFlag = 0x14;
+    globalCtx->unk_1887F = 2;
+}
+
+// Gameplay_TriggerRespawn ?
+// Used for void by Wallmaster, Deku Shrine doors. Also used by Player, Kaleido, DoorWarp1
+void func_80169FDC(GameState* gameState) {
+    func_80169F78(gameState);
+}
+
+// Used by Kankyo to determine how to change the lighting, e.g. for game over.
+s32 func_80169FFC(GameState* gameState) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+
+    return globalCtx->roomCtx.currRoom.mesh->type0.type != 1;
+}
+
+s32 FrameAdvance_IsEnabled(GameState* gameState) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+
+    return globalCtx->frameAdvCtx.enabled != 0;
+}
+
+// Unused, unchanged from OoT, which uses it only in one Camera function.
+/**
+ * @brief Tests if \p actor is a door and the sides are different rooms.
+ *
+ * @param[in] gameState GameState, promoted to globalCtx inside.
+ * @param[in] actor Actor to test.
+ * @param[out] yaw Facing angle of the actor, or reverse if in the back room.
+ * @return true if \p actor is a door and the sides are in different rooms, false otherwise
+ */
+s32 func_8016A02C(GameState* gameState, Actor* actor, s16* yaw) {
```

Comment:
Is this style of docs for some kind of external use? It seems slightly harder to read in game. Why are `params` mentioned? That's a common variable name that doesn't seem to be present in this function. What are the `\p` slashes? Does return need the @? afaik this is the first time seeing this style

Reply:
This is Doxygen. VSCode will render it nicely if you hover over a function, and you can set it up to produce html (TMC have already done this: https://discord.com/channels/688807550715560050/716062722302804009/928167610011512832). There's already a few functions documented like this (EnMag PR, z_actor has some), and we're most likely going to be going this way in future.

Reply:
Yeah I figured it was something like that. It's fine then